### DISCUSSION TITLE
Where is the documentation ?

### DISCUSSION THREAD

**Discussion by TechieGeeke** [efroemling/ballistica#448] [Q&A] [answered]
Where can I find the docs for modding about ba module? The [old one ](https://github.com/efroemling/ballistica/blob/master/docs/ba_module.md) doesnt work anymore....

**Comment by IvanPragma** [comment, marked as answer]
#389
But you can still view the outdated version of docs: https://github.com/efroemling/ballistica/blob/f2206923985555350265ba908618ef0135edf13e/docs/ba_module.md

**Reply by TechieGeeke** [reply]
Isn't the latest version of documentation not yet released?

**Reply by TechieGeeke** [reply]
Is there a function to callback when someone send a message in the chat? Or is there a function to check if the user is a player or server host?

**Reply by IvanPragma** [reply]
`ba._hooks.filter_chat_message`. To check whether a user is a server host, you can check its `client_id`. The host `client_id` will always be `-1`

**Reply by TechieGeeke** [reply]
> `ba._hooks.filter_chat_message`. To check whether a user is a server host, you can check its `client_id`. The host `client_id` will always be `-1`

Thanks for the reply; I am very new to bs API. So, may I ask, can I get the client id from the server hostname? Some hosts have the name `Server122121` while others have modified custom names. The thing I want to check is whether the last messaged user is the server host or not, like:

```python
cm = _ba.get_chat_messages()[-1] #Last message be like: "Server12356: Hello users"
splitted_cm = cm.split(": ") #After splitting: ["Server12356","Hello users"]
if  if splitted_cm[0] == {SERVER HOST}: #To check if "Server12356" == host!
     #do something..
```

What should I replace instead of `{SERVER HOST}` if there's some way...

**Reply by TechieGeeke** [reply]
Also could you explain how `ba._hooks.filter_chat_message` works?

**Reply by IvanPragma** [reply]
> What should I replace instead of `{SERVER HOST}` if there's some way...

Use `_ba.get_v1_account_name()` to get host name. But this is not a very good way, because at the moment `account_name` can be easily spoofed. Check `client_id` instead of `account_name`. To get `client_id` use `sessionplayer.inputdevice.client_id`

> Also could you explain how ba._hooks.filter_chat_message works?

When the messages arrive, the c++ level will call cached `ba._hooks.filter_chat_message`, passing the sender's `client_id` and message there.

**Reply by TechieGeeke** [reply]
> To get client_id use sessionplayer.inputdevice.client_id

When I did that it gave me an error:
 ````
print(sessionplayer.inputdevice.client_id)
 NameError: name 'sessionplayer' is not defined

**Reply by IvanPragma** [reply]
🤦‍♂️

**Reply by TechieGeeke** [reply]
> 🤦‍♂️

I'm noob  😁

**Reply by TechieGeeke** [reply]
@IvanPragma You on discord?

**Reply by IvanPragma** [reply]
I rarely check discord, telegram @ms_ept

**Reply by TechieGeeke** [reply]
> I rarely check discord, telegram @ms_ept

Please discord! PC ppl cant have telegram without phone for some reason...

**Reply by IvanPragma** [reply]
IvanPragma#9302

**Reply by TechieGeeke** [reply]
> IvanPragma#9302

Aight! I have send you a friend request. My name is `Evo Pro#4879`

**Reply by TechieGeeke** [reply]
@IvanPragma check discord please..

**Reply by TechieGeeke** [reply]
@IvanPragma Discord 😁

**Reply by TechieGeeke** [reply]
@IvanPragma Discord please :)

**Comment by efroemling** [comment]
Ok, I just got the revised docs generation wired up: https://ballistica.net/docs
This will update itself nightly (about 5am California time) based on the contents of this repo.
It will also keep different api versions separate so it'll be possible to look at stable api docs even if things are changing on newer dev versions of the game.
A huge thanks to @Dliwk for wiring up the base docs-generation code to use [pdoc](https://pdoc.dev). There's certainly more we can do to polish things up, but this gives us some nice features like search and covers more modules than just ba and is a great starting point.

**Reply by Dliwk** [reply]
Seems like you forgot to make /docs url 'public'. :)
![изображение](https://user-images.githubusercontent.com/47295209/180089994-b15332f7-655c-47ee-9965-10af17a2c48d.png)

**Reply by TechieGeeke** [reply]
Woah! Indeed great work man. This is so useful and so neat and clean to read. Moreover, it's lag-free and easily accessible. Great thanks for taking the initiative to build this. Thanks a lot, @Dliwk and @efroemling . Always looking forward for more updates

**Reply by efroemling** [reply]
> Why don't you also fix the other links in 'https://www.froemling.net' such as 'https://www.froemling.net/docs/bombsquad-python-api'? It would be more useful to people who come by searching on the web.

Good call; I'll do that now 👍🏻

**Reply by efroemling** [reply]
Ok, looks like I've already got a red notice at the top of each of those old pages pointing to the wiki here (which has links to the python api/etc). Do you think I should do something more than that?

**Reply by TechieGeeke** [reply]
> > Why don't you also fix the other links in 'https://www.froemling.net' such as 'https://www.froemling.net/docs/bombsquad-python-api'? It would be more useful to people who come by searching on the web.
> 
> Good call; I'll do that now 👍🏻

Actually I thought, you guys were busy and have many more things to do than to work on links and wikis. Thats why I deleted that lol

**Comment by efroemling** [comment]
Oops I certainly did 🤦🏼‍♂️. I set up my new-page template to be admin-only by default to prevent me from accidentally making sensitive stuff public. But instead I'll probably accidentally create all public stuff as admin-only to start with. 😬.  Check back in like 2 min and this should be fixed.

**Comment by efroemling** [comment]
Ok; should be fixed. Holler if not.
Also, if anyone sees places in the wiki/etc. that are still pointing at the old docs, please let me know or go ahead and update them to point at https://ballistica.net/docs
